sw/device/tests/gpio_pinmux_test.c

```diff
@@ -0,0 +1,60 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/dif/dif_gpio.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/json/command.h"
+#include "sw/device/lib/testing/json/gpio.h"
+#include "sw/device/lib/testing/json/pinmux_config.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_flow_control.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/lib/testing/test_framework/ujson_ottf.h"
+#include "sw/device/lib/ujson/ujson.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG(.want_flow_control = true);
+
+static dif_gpio_t gpio;
+static dif_pinmux_t pinmux;
+
+status_t command_processor(ujson_t *uj) {
+  while (true) {
+    test_command_t command;
+    TRY(ujson_deserialize_test_command_t(uj, &command));
```

Comment:
Does this macro hide a return or break or something for the non-error case? How do we reach the test status reporting?

Reply:
The `TRY` macro early-exits for error conditions.

I assume that by "test status reporting", you're referring to the writes to pseudo-registers after `test_main` exits.  That _never_ happens during successful execution of this type of test (this test program never exits - it's always waiting for a next command).

The pass/fail criteria of this test are evaluated by the test_harness program.

Reply:
Ahh, in that case, can you return some sort of error below instead of OK_STATUS? (so there is at least something showing that is supposed to be unreachable)